src/app/dataset-view/additional-components/flows-component/flows.component.ts

```diff
@@ -0,0 +1,159 @@
+import { MaybeNull, MaybeUndefined } from "src/app/common/app.types";
+import { BaseComponent } from "src/app/common/base.component";
+import { NavigationEnd, Router, RouterEvent } from "@angular/router";
+import { requireValue } from "src/app/common/app.helpers";
+import ProjectLinks from "src/app/project-links";
+import { NavigationService } from "src/app/services/navigation.service";
+import { DatasetViewTypeEnum } from "../../dataset-view.interface";
+import { SettingsTabsEnum } from "../dataset-settings-component/dataset-settings.model";
+import { FilterByInitiatorEnum } from "./components/flows-table/flows-table.types";
+
+@Component({
+    selector: "app-flows",
+    templateUrl: "./flows.component.html",
+    styleUrls: ["./flows.component.scss"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FlowsComponent extends BaseComponent implements OnInit {
+    @Input() public datasetBasics: DatasetBasicsFragment;
+    @Output() onPageChangeEmit = new EventEmitter<number>();
+    public searchFilter = "";
+    public tileWidgetData$: Observable<MaybeUndefined<FlowConnectionDataFragment>>;
+    public flowConnectionData$: Observable<MaybeUndefined<FlowConnectionDataFragment>>;
+    public filterByStatus: MaybeNull<FlowStatus> = null;
+    public filterByInitiator = FilterByInitiatorEnum.All;
+    public searchByAccountName = "";
+    public currentPage = 1;
+    public readonly WIDGET_FLOW_RUNS_PER_PAGE: number = 150;
+    public readonly TABLE_FLOW_RUNS_PER_PAGE: number = 15;
+    public readonly FlowStatus: typeof FlowStatus = FlowStatus;
+
+    constructor(
+        private flowsService: DatasetFlowsService,
+        private router: Router,
+        private navigationService: NavigationService,
+    ) {
+        super();
+    }
+
+    ngOnInit(): void {
+        this.getTileWidgetData();
+        this.datasetFlowListByPage();
+        this.trackSubscriptions(
+            this.router.events
+                .pipe(
+                    filter((event) => event instanceof NavigationEnd),
+                    map((event) => event as RouterEvent),
+                )
+                .subscribe(() => this.datasetFlowListByPage()),
+        );
+    }
+
+    public getFlowConnectionData(
+        page: number,
+        filterByStatus?: MaybeNull<FlowStatus>,
+        filterByInitiator?: MaybeNull<InitiatorFilterInput>,
+    ): void {
+        this.flowConnectionData$ = this.flowsService.datasetFlowsList({
+            datasetId: this.datasetBasics.id,
+            page: page - 1,
+            perPage: this.TABLE_FLOW_RUNS_PER_PAGE,
+            filters: { byStatus: filterByStatus, byInitiator: filterByInitiator },
+        });
+    }
+
+    public getTileWidgetData(): void {
+        this.tileWidgetData$ = this.flowsService.datasetFlowsList({
+            datasetId: this.datasetBasics.id,
+            page: 0,
+            perPage: this.WIDGET_FLOW_RUNS_PER_PAGE,
+            filters: {},
+        });
+    }
```

Comment:
It's fine for now, but we will have to do a better indermediate service when we will add real-time updates

Reply:
Ok